### PR TITLE
Closes #2191: Bigint Groupby Benchmark

### LINKED ISSUE #2191: bigint groupby benchmark
Add benchmark to track groupby on bigint arrays

### PR DESCRIPTION
**Pull request by pierce314159**
This PR (closes #2191):
- Adds bigint groupby benchmark
- Moves bigint benchmarks to their own suite
- Adds a small optimization in `bigint_from_uint_arrays` that avoids sending unnecessary uint arrays to server when it's known they won't fit in `max_bits`. For example, if `max_bits=64`, there's no need to send 4 uint arrays to server, since the top 3 uint arrays won't be used.

Here are the perf numbers with comm=None and quick compile. I don't see a way to improve perf easily. To me, it does seem to track to be about the amount of time of an int64 groupby with twice as many arrays plus the cost of running`bigint_to_uint_arrays`. Which is pretty much exactly what we do

```
% ./benchmarks/run_benchmarks.py bigint_groupby -t 1
array size = 100,000,000
number of trials =  1
>>> arkouda bigint groupby
numLocales = 1, N = 100,000,000
1-array Average time = 14.3902 sec
1-array Average rate = 0.1036 GiB/sec
2-array Average time = 23.6362 sec
2-array Average rate = 0.0630 GiB/sec
8-array Average time = 28.2082 sec
8-array Average rate = 0.0528 GiB/sec
16-array Average time = 28.9222 sec
16-array Average rate = 0.0515 GiB/sec

% ./benchmarks/run_benchmarks.py bigint_groupby -t 1 --max-bits=64
array size = 100,000,000
number of trials =  1
>>> arkouda bigint groupby
numLocales = 1, N = 100,000,000
1-array Average time = 11.8893 sec
1-array Average rate = 0.0627 GiB/sec
2-array Average time = 19.1546 sec
2-array Average rate = 0.0389 GiB/sec
8-array Average time = 19.7305 sec
8-array Average rate = 0.0378 GiB/sec
16-array Average time = 20.4340 sec
16-array Average rate = 0.0365 GiB/sec
```

### CODE REVIEW

File: benchmarks/graph_infra/arkouda-bigint.graph [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+graphkeys: bigint AND GiB/s, bigint OR GiB/s, bigint SHIFT GiB/s
+files: bigint_bitwise_binops.dat, bigint_bitwise_binops.dat, bigint_bitwise_binops.dat
+graphtitle: Bigint Bitwise Binops Performance
+ylabel: Performance (GiB/s)
+
+perfkeys: Average bigint AND rate =, Average bigint OR rate =, Average bigint SHIFT rate =
+graphkeys: bigint AND GiB/s, bigint OR GiB/s, bigint SHIFT GiB/s
+files: bigint_bitwise_binops.dat, bigint_bitwise_binops.dat, bigint_bitwise_binops.dat
+graphtitle: Bigint Bitwise Binops Performance
+ylabel: Performance (GiB/s)
```
Comment (ronawho) [inline benchmarks/graph_infra/arkouda-bigint.graph:23]: Duplicate "Bigint Bitwise Binops Performance" graphs?

**Review by ronawho — APPROVED** [review]
Looks good to me except for duplicate graph. @bmcdonald3 could you check 16-node-cs-hdr perf?

**Review by Ethan-DeBandi99 — APPROVED** [review]
Looks good. Agree with Elliot that it appears there are duplicate graphs, but otherwise I don't see any issues.

**Comment by bmcdonald3** [comment]
Here are some numbers with a much smaller problem size on 16-node-cs-hdr:
```
$ python3 benchmarks/bigint_groupby.py <hostname> 5555 -n=100000
array size = 100,000
number of trials =  1
>>> arkouda bigint groupby
numLocales = 16, N = 1,600,000
1-array Average time = 1.8021 sec
1-array Average rate = 0.0132 GiB/sec
2-array Average time = 2.1310 sec
2-array Average rate = 0.0112 GiB/sec
8-array Average time = 2.1654 sec
8-array Average rate = 0.0110 GiB/sec
16-array Average time = 2.2135 sec
```
and here is that ran with a 10x larger problem size (100x below default):
```
$ python3 benchmarks/bigint_groupby.py <hostname> 5555 -n=1000000
array size = 1,000,000
number of trials =  1
>>> arkouda bigint groupby
numLocales = 16, N = 16,000,000
1-array Average time = 16.4960 sec
1-array Average rate = 0.0145 GiB/sec
2-array Average time = 15.3211 sec
2-array Average rate = 0.0156 GiB/sec
8-array Average time = 15.2540 sec
8-array Average rate = 0.0156 GiB/sec
16-array Average time = 15.1145 sec
16-array Average rate = 0.0158 GiB/sec
```

Trying the smaller problem size at 8 locales:
```
$ python3 benchmarks/bigint_groupby.py <hostname> 5555 -n=100000
array size = 100,000
number of trials =  1
>>> arkouda bigint groupby
numLocales = 8, N = 800,000
1-array Average time = 1.7891 sec
1-array Average rate = 0.0067 GiB/sec
2-array Average time = 1.9786 sec
2-array Average rate = 0.0060 GiB/sec
8-array Average time = 1.9403 sec
8-array Average rate = 0.0061 GiB/sec
16-array Average time = 1.9641 sec
16-array Average rate = 0.0061 GiB/sec
```

So, based on these numbers, the current default problem size would take ~2 hours to complete on 16-node-cs-hdr.

**Comment by ronawho** [comment]
Ok, based on those numbers I'd be inclined to drop the problem size down to `10**6` for now and then look to optimize (where I'm hoping aggregation may be all that's needed, but there could easily be other overheads)

**Review by joshmarshall1 — APPROVED** [review]
Looks good aside from what's already been mentioned

**Pull request merged**